flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -185,6 +199,10 @@ public ResourceManager(
 		this.jmResourceIdRegistrations = new HashMap<>(4);
 		this.taskExecutors = new HashMap<>(8);
 		this.taskExecutorGatewayFutures = new HashMap<>(8);
+
+		this.defaultTaskManagerMemoryMB = ConfigurationUtils.getTaskManagerHeapMemory(flinkConfig).getMebiBytes();
+		this.numberOfTaskSlots = flinkConfig.getInteger(TaskManagerOptions.NUM_TASK_SLOTS);
+		this.slotsPerWorker = updateTaskManagerConfigAndCreateWorkerSlotProfiles(this.flinkConfig, defaultTaskManagerMemoryMB, numberOfTaskSlots);
```

Comment:
Should this also happen in `StandaloneResourceManager`?
It just looks as if this change was kept out of `StandaloneResourceManager` on purpose or there was another reason not to do it like this before?

Reply:
ok, I see it is actually not used in `StandaloneResourceManager`.
Yarn and mesos RMs could extend an abstract `ResourceManagerWithSlotsPerWorker` but it will probably complicate things.
It will probably need to change with the dynamic memory slicing anyways.

Reply:
Yes, the background of this calculating `slotsPerWorker` on RM is that, we need to know the resource profiles of slots for `PendingTaskManagerSlot` before TM is started and registered. For standalone, we don't have any pending task manager slot because RM cannot actively start any TM.